### PR TITLE
REGoth Engine Resources (Shaders, etc) loading

### PR DESCRIPTION
**Pull request by ataulien**
This PR adds modules which can find a `content`-directory where resources needed by the engine can be stored and load them. Also it refines the resource management inside the base `REGothEngine`-class and should fix the "black textures"-bug for good.

The main change however is the new shader applied to most of the worlds geometry. It has less fancy features than the default one (ie. no PBR), but supports vertex lighting and masked textures.

### CODE REVIEW

File: content/shaders/World.bsl [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+  code
+  {
+    [alias(gAlbedoTex)] SamplerState gAlbedoSamp;
+
+    Texture2D gAlbedoTex = white;
+
+    struct ZenVertexInput
+    {
+      #if POSITION_2D
```
Comment (mworchel) [inline content/shaders/World.bsl]: Don't you need a `variation` to do this or is it not yet used?

Comment (ataulien) [reply content/shaders/World.bsl]: I don't know, this was copied from the original vertex input structure. I haven't read that deep into the shader language yet. However, I've seen some variations being defined in one of the included headers.

File: src/REGothEngine.cpp [hunk 2]
```diff
@@ -181,12 +222,11 @@ int ::REGoth::main(REGothEngine& regoth, int argc, char** argv)
 
   regoth.loadCachedResourceManifests();
 
+  regoth.setShaders();
   regoth.setupInput();
   regoth.setupMainCamera();
   regoth.setupScene();
 
-  regoth.saveCachedResourceManifests();
```
Comment (mworchel) [inline src/REGothEngine.cpp]: You forgot to remove the same line in `src/main.cpp`, so the `REGoth` target doesn't compile.

**Comment by mworchel** [comment]
Add targets except for the `REGoth` one compile but no textures are shown (everything is black). I deleted the cache folder and even deleted the whole build directory and recompiled. Tested with the `WaynetTester`.

**Comment by ataulien** [comment]
I can reproduce the black textures. Damn, I was hoping this was an issue with the standard shader and/or resource manifest. I guess I'll have to dig deeper into why that happens.

There is also a huge slowdown caused by saving the resource manifest so often, so I need to get rid of that too.

**Comment by ataulien** [comment]
Black Textures should be resolved now. @mworchel can you try again, please?

**Comment by mworchel** [comment]
> 
> 
> Black Textures should be resolved now. @mworchel can you try again, please?

I tried it and the issue seems to be resolved. No more black textures.

**Comment by ataulien** [comment]
Awesome! I'll merge it then.

**Pull request merged**